tests/RobotFramework/tasks/E2E_Tutorials/connect_to_c8y_send_measurements.robot

```diff
@@ -0,0 +1,173 @@
+
+Create Device Certificate
+    [Documentation]    Create a self-signed certificate for the device.
+    Execute Command    sudo tedge cert create --device-id ${DEVICE_SN}
+    Log    Created device certificate for ${DEVICE_SN}
+
+Check Device Certificate
+    [Documentation]    Verify the contents of the device certificate.
+    ${output}=    Execute Command    sudo tedge cert show
+    Log    ${output}
+    Should Contain    ${output}    Device certificate: /etc/tedge/device-certs/tedge-certificate.pem
+    Should Contain    ${output}    Subject: CN=${DEVICE_SN}, O=Thin Edge, OU=Test Device
+    Should Contain    ${output}    Issuer: CN=${DEVICE_SN}, O=Thin Edge, OU=Test Device
+    Should Contain    ${output}    Valid from:
+    Should Contain    ${output}    Valid up to:
+    Should Contain    ${output}    Thumbprint:
+
+Upload Device Certificate
+    [Documentation]    Upload the device certificate to Cumulocity IoT.
+    ${output}     Execute Command    sudo env C8YPASS\='${C8Y_CONFIG.password}' tedge cert upload c8y --user ${C8Y_CONFIG.username}
+    Log    ${output}
+    Should Contain    ${output}    Certificate uploaded successfully.
+    Log    Uploaded device certificate for ${DEVICE_SN}
+
+Connect to Cumulocity
+    [Documentation]    Connect the device to Cumulocity IoT.
+    ${output}=    Execute Command    sudo tedge connect c8y
+    Log    ${output}
+    Should Contain    ${output}    Checking if systemd is available.
+    Should Contain    ${output}    Checking if configuration for requested bridge already exists.
+    Should Contain    ${output}    Validating the bridge certificates.
+    Should Contain    ${output}    Creating the device in Cumulocity cloud.
+    Should Contain    ${output}    Saving configuration for requested bridge.
+    Should Contain    ${output}    Restarting mosquitto service.
+    Should Contain    ${output}    Awaiting mosquitto to start. This may take up to 5 seconds.
+    Should Contain    ${output}    Enabling mosquitto service on reboots.
+    Should Contain    ${output}    Successfully created bridge connection!
+    Should Contain    ${output}    Sending packets to check connection. This may take up to 2 seconds.
+    Should Contain    ${output}    Connection check is successful.
+    Should Contain    ${output}    Checking if tedge-mapper is installed.
+    Should Contain    ${output}    Starting tedge-mapper-c8y service.
+    Should Contain    ${output}    Persisting tedge-mapper-c8y on reboot.
+    Should Contain    ${output}    tedge-mapper-c8y service successfully started and enabled!
+    Should Contain    ${output}    Enabling software management.
+    Should Contain    ${output}    Checking if tedge-agent is installed.
+    Should Contain    ${output}    Starting tedge-agent service.
+    Should Contain    ${output}    Persisting tedge-agent on reboot.
+    Should Contain    ${output}    tedge-agent service successfully started and enabled!
```

Comment:
I don't think it is worth explicitly checking every message here...as generally there is no explicit contract between console output and the user (except for "conveying useful information")...The official contract is that the command should exit with an exit code of `0` if everything worked as expected.

You could add an additional connection test, `tedge connect c8y --test` to verify that the connection is working if you wanted to verify it (but I'm pretty sure we already have system tests for that part).

Reply:
Please have in mind that the main purpose of the tasks is not to test the feature because we do have tests that are doing that, with the tasks I am checking the documentation correctness and in the documentation (https://thin-edge.github.io/thin-edge.io/start/connect-c8y/) this content is stated as it should exist

Reply:
Yes, but we have to remember that the "accurate documentation" does not mean that every single console output needs to be 100% accurate, as the exact output is not important, it is more important that the "commands" and the outcome of the commands are accurate.

If we do verbatim comparisons of the console output, it will just introduce more problems to keep everything in sync rather than actually providing value to the user.

Reply:
at least I will get alerted that something has been changed, which is my aim

Reply:
ok, though I don't think it provides much value.